src/rgw/rgw_notify.cc

```diff
@@ -667,24 +671,23 @@ void tags_from_attributes(const req_state* s, rgw::sal::Object* obj, KeyValueMap
 }
 
 // populate event from request
-void populate_event_from_request(const reservation_t& res, 
+static inline void populate_event(reservation_t& res,
         rgw::sal::Object* obj,
         uint64_t size,
         const ceph::real_time& mtime, 
         const std::string& etag, 
         const std::string& version, 
         EventType event_type,
         rgw_pubsub_s3_event& event) {
-  const auto s = res.s;
   event.eventTime = mtime;
-  event.eventName = to_event_string(event_type);
-  event.userIdentity = s->user->get_id().id;    // user that triggered the change
-  event.x_amz_request_id = s->req_id;          // request ID of the original change
-  event.x_amz_id_2 = s->host_id;               // RGW on which the change was made
+  event.eventName = to_string(event_type);
```

Comment:
```suggestion
  event.eventName = to_event_string(event_type);
```

in the published event we need the name without the "s3::" prefix

Reply:
@mattbenjamin please fix this. this is failing all tests